src/Model/Event.java

```diff
@@ -36,6 +36,8 @@
     protected GregorianCalendar date = null;
 
     private static Pattern dateRegex = Pattern.compile("(\\d\\d)/(\\d\\d)/(\\d\\d\\d\\d)T(\\d\\d):(\\d\\d):(\\d\\d)Z");
+    
+    protected SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy'T'hh:mm:ss'Z'");
```

Comment:
Why is this protected instead of private?  Also, date formatters are expensive to create, so it should be made `static`.  However, they are not thread-safe.  Quoting from the documentation:

> **Synchronization**
> 
> Date formats are not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.

https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html

I recommend placing it inside of a `ThreadLocal<>`:  https://docs.oracle.com/javase/8/docs/api/java/lang/ThreadLocal.html